### ISSUE TITLE
C++ Builder complains redefinition of stack

### ISSUE THREAD

**Issue by Dediggefedde** [kunitoki/LuaBridge3#50]
I would like to use LuaBride3 in a c++ 32-bit windows project with the Embarcadero C++ Builder (Community).

Just including "LuaBridge3.h" stops compilation in line 2176: `redefinition of 'Stack<char,void>'`
```c++
template <>
struct Stack<int8_t>
{
```
Previous line here: 2128
```c++
template <>
struct Stack<char>
{
```

Example code (VCL application):
```c++
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "mainform.h"

#include "lua5.4/lua.hpp"
#include "LuaBridge/LuaBridge.h"

#include <iostream>
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
	: TForm(Owner)
{
//
}
```

I assume C++ Builder treats `int8_t` the same as `char`.
Do you know of a flag or project setting I can change to fix the error?

**Comment by kunitoki** [comment]
I cannot think of a way to fix it unless explicitly exclude one of the two definitions for borlandc. could you try adding a macro excluding borlandc compiler from the int8_t specialization:

```cpp
#if !defined(__BORLANDC__)
template <>
struct Stack<int8_t>
{
...
};
#endif
```

if that works, i can add it to the library

**Comment by kunitoki** [comment]
@Dediggefedde does this work for you ?

**Comment by Dediggefedde** [comment]
@kunitoki 
thank you for the suggestion! 
I'm sorry for my delayed response. 

I tried your suggestion, changing the file _/luabridge3/detail/Stack.h_ in line 247 and 280 to use the if-directive you mentioned.
including the header file compiles now, but now the linker complains for a test-program for unresolved objects like  `__stack_chk_fail`:

```c++

#include <vcl.h>
#pragma hdrstop

#include "mainform.h"

#include "lua/lua.hpp"
#include "luabridge3/LuaBridge.h"

#include <iostream>
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;
__fastcall TForm1::TForm1(TComponent* Owner)
	: TForm(Owner)
{
	lua_State* luaState = luaL_newstate();
}

```
![grafik](https://user-images.githubusercontent.com/4397138/229358348-5383db20-4f36-4b41-a1aa-044166ef32f8.png)

I'm also still a bit unsure if I use all the required library files. I'm using "liblua.a" "liblua5.4.a" and "liblua5.4-c++.a" in a 64bit project at the moment.

**Comment by kunitoki** [comment]
That linker issue is a different problem, seems like you are linking a library made with mingw, have you tried compiling lua with borlandc instead ?

**Comment by kunitoki** [comment]
Can you also try with this instead of the ifdef ?

```cpp
template <>
struct Stack<int8_t, std::enable_if_t<! std::is_same_v<char, int8_t>>>
{
```

**Comment by Dediggefedde** [comment]
> Can you also try with this instead of the ifdef ?
> 
> ```c++
> template <>
> struct Stack<int8_t, std::enable_if_t<! std::is_same_v<char, int8_t>>>
> {
> ```

With this code, I get an error: `stddef.h(250): no type named 'type' in 'std::enable_if<false, void>'; 'enable_if' cannot be used to disable this declaration`
<br>

> That linker issue is a different problem, seems like you are linking a library made with mingw, have you tried compiling lua with borlandc instead ?

Yes, thank you! Sorry, I'm a bit inexperienced with compiling external libraries.
I used a pretty extensive instruction to use WSL to generate the lua library, since I didn't find anything for C++ Builder.
Now I checked and noticed C++ Builder compiles me the `lib` and `a` file with two clicks in windows. ^^' 
And now everything compiles and the test-examples also work.

Is there a reason `#if !defined(__BORLANDC__)` should be avoided in favor of `enable_if_t`? 
Otherwise, I see the problem as solved.
